Pass `allItems` and `currentItems` to `_pagination.paginate()` (#1100)
sindresorhus/got: readme.md

###### \_pagination.paginate

Type: `Function`\
Default: [`Link` header logic](source/index.ts)

The function takes three arguments: `response` (response object), `allItems` (array) and `currentItems` (array).
It should return an object representing Got options pointing to the next page. If there are no more pages, `false` should be returned.

> For example, if you want to stop when the response contains less items than expected, you should use `(response, allItems, currentItems) => currentItems.length < response.request.options.searchParams.entriesPerPage ? false : { url: getNextLink(response, allItems, currentItems) }`.

Reviewer: This needs to be a

```
block
```

Not a `single line` example.

###### \_pagination.paginate

Type: `Function`\

> Default: `(response, allItems, currentItems) => getLinkFromHeaders(response)` (find details [here](source/index.ts))

Reviewer: This change is unnecessary

Author: Okay, I restored it but changed the sentence afterwards to make clear how the signature of the function should be. This should make it easier for users without diving into the code first.

A function that returns an object representing Got options pointing to the next page. If there are no more pages, `false` should be returned.

> For example, if you want to stop when the response contains less items that expected, you should use `(response, allItems, currentItems) => currentItems.length < LIMIT ? false : { url: getNextLink(response, allItems, currentItems) }`.

Reviewer: the `LIMIT` should be a `response.request.options.searchParams.entriesPerPage`

Reviewer: `that expected` -> `than expected`

Author: Okay, it's updated.

Type: `Function`\
Default: [`Link` header logic](source/index.ts)

The function takes three arguments:
- `response` (the current response object)
- `allItems` (an array of the emitted items)
- `currentItems` (items from the current response).

It should return an object representing Got options pointing to the next page. If there are no more pages, `false` should be returned.

For example, if you want to stop when the response contains less items than expected, you can use something like this:

```js

> (response, allItems, currentItems) => {

Reviewer 2: I would prefer the example to be fully runnable, meaning that it includes the import statement and `got(…)`.

Author: I just extended the example.

The function takes three arguments: `response` (response object), `allItems` (array) and `currentItems` (array).
It should return an object representing Got options pointing to the next page. If there are no more pages, `false` should be returned.

For example, if you want to stop when the response contains less items than expected, you should use:
```
(response, allItems, currentItems) => {

> 	if (currentItems.length < response.request.options.searchParams.entriesPerPage) {

Reviewer: ```suggestion
	const entriesPerPage = response.request.options.searchParams.get('entriesPerPage');

	if (entriesPerPage !== undefined && currentItems.length < entriesPerPage) {
```

###### \_pagination.filter

Type: `Function`\
Default: `(item, allItems, currentItems) => true`

Checks whether the item should be emitted or not. `currentItems` is an array of the emitted of items of the current response and `allItems` is an error of _all_ the previously emitted items.

###### \_pagination.shouldContinue

Type: `Function`\
Default: `(item, allItems, currentItems) => true`

> Checks whether the pagination should continue. `currentItems` is an array of the emitted of items of the current response and `allItems` is an error of _all_ the previously emitted items.

Reviewer: Error? That's incorrect.

```suggestion
Checks whether the pagination should continue. `allItems` is an array of the all emitted items, while `currentItems` is an array of the current response items.
```

Author: Of course, it was just a typo. Good catch 😊

> For example, if you need to stop **before** emitting an entry with some flag, you should use `(item, allItems, currentItems) => !item.flag`. If you want to stop **after** emitting the entry, you should use `(item, allItems) => allItems.some(entry => entry.flag)` instead.

Reviewer: You should also include an example of your use case.
